### PR TITLE
ci: add multus integration test

### PR DESCRIPTION
**Pull request by leseb**
**Description of your changes:**

This new integration test will deploy a cluster with multus enabled. It
will be comprised of two network interfaces for ceph public and cluster
communications.
For now, it only deploys a Ceph cluster up to the OSDs.

Closes: https://github.com/rook/rook/issues/9784
Signed-off-by: Sébastien Han <seb@redhat.com>

**Which issue is resolved by this Pull Request:**
Resolves #

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/development-flow.html#commit-structure).
- [ ] **Skip Tests for Docs**: If this is only a documentation change, add the label `skip-ci` on the PR.
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/development-flow.html#submitting-a-pull-request)
- [ ] [Pending release notes](https://github.com/rook/rook/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

File: deploy/examples/cluster-test-multus.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
+#################################################################################################################
```
Comment (travisn) [inline deploy/examples/cluster-test-multus.yaml:1]: How about naming the file `cluster-multus-test.yaml`? The test suffix is more consistent with other yaml examples this way.

File: .github/workflows/canary-integration-test.yml [hunk 2]
```diff
@@ -908,3 +908,48 @@ jobs:
       if: failure() && github.event_name == 'pull_request'
       uses: mxschmitt/action-tmate@v3
       timeout-minutes: 60
+
+  multus-cluster-network:
```
Comment (travisn) [inline .github/workflows/canary-integration-test.yml:912]: The interesting part of this test will be when we can test a volume with multus. After csi 3.6.1 is integrated we can add that test?

**Review by travisn — APPROVED** [review]
Can we check for the running holder pod? Or in a follow-up?

**Review by BlaineEXE — APPROVED** [review]

**Comment by leseb** [comment]
> Can we check for the running holder pod? Or in a follow-up?

It's somewhat being done by this line: https://github.com/rook/rook/pull/10048/files#diff-2fd42145a1763de4631a65e88fc4740083d8af2a9b7052bd13be79c372a54701L92 but can certainly be improved.

**Pull request merged**